pcdsdevices/pim.py

```diff
@@ -30,39 +28,197 @@ class PIMMotor(InOutRecordPositioner):
-        the motor prefix
+        The EPICS base PV of the detector. If None, it will be attemptted to be
+        inferred from `prefix`
+
+    prefix_zoom : str, optional
+        The EPICS base PV of the zoom motor. If None, it will be attempted to
+        be inferred from `prefix`
     """
-    detector = FCpt(PCDSAreaDetector, "{self._prefix_det}", kind='normal')
-    tab_whitelist = ["detector"]
-
-    def __init__(self, prefix, *, name, prefix_det=None, **kwargs):
-        # Infer the detector PV from the motor PV
-        if not prefix_det:
-            self._section = prefix.split(":")[0]
-            self._imager = prefix.split(":")[1]
-            self._prefix_det = "{0}:{1}:CVV:01".format(
-                self._section, self._imager)
-        else:
+
+    _prefix_start = ''
+
+    state = Cpt(PIM_Y, '', kind='normal')
+    zoom_motor = FCpt(IMS, '{self._prefix_zoom}', kind='normal')
+    detector = FCpt(PCDSAreaDetector, '{self._prefix_det}', kind='normal')
+
+    tab_whitelist = ['y_motor', 'zoom_motor', 'detector']
+
+    def infer_prefix(self, prefix):
+        """Pulls out the first two segments of the prefix PV, if not already
+           done"""
+        if not self._prefix_start:
+            self._prefix_start = '{0}:{1}:'.format(prefix.split(':')[0],
+                                                   prefix.split(':')[1])
+
+    @property
+    def prefix_start(self):
+        """Returns the first two segments of the prefix PV."""
+        return str(self._prefix_start)
+
+    @property
+    def removed(self):
+        """Returns ``True`` if the yag and diode are removed from the beam."""
+        return self.state.removed
+
+    @property
+    def inserted(self):
+        """Returns ``True`` if yag or diode are inserted."""
+        return self.state.inserted
+
+    def insert(self, moved_cb=None, timeout=None, wait=False):
+        """Moves the diode into the beam."""
+        return self.state.insert(moved_cb=moved_cb, timeout=timeout,
+                                 wait=wait)
+
+    def remove(self, moved_cb=None, timeout=None, wait=False):
+        """Moves the diode out of the beam."""
+        return self.state.remove(moved_cb=moved_cb, timeout=timeout,
+                                 wait=wait)
+
+    def __init__(self, prefix, *, name, prefix_det=None, prefix_zoom=None,
+                 **kwargs):
+        self.infer_prefix(prefix)
+
+        # Infer the detector PV from the base prefix
+        if prefix_det:
             self._prefix_det = prefix_det
+        else:
+            self._prefix_det = self.prefix_start+'CVV:01'
+
+        # Infer the zoom motor PV from the base prefix
+        if prefix_zoom:
+            self._prefix_zoom = prefix_zoom
+        else:
+            self._prefix_zoom = self.prefix_start+'CLZ:01'
+
+        super().__init__(prefix, name=name, **kwargs)
+        self.y_motor = self.state.motor
+
+
+class PIM_withFocus(PIM):
```

Comment:
The python convention here would be something like:
```suggestion
class PIMWithFocus(PIM):
```
But that looks ugly so idk

Reply:
Both are less than ideal - the other alternative could be `PimWithFocus`, but I like that even less. I suppose my vote would go with `PIMWithFocus` even if the camel-case name loses a bit of meaning.

Reply:
Updated in new commit.